plugins/module_utils/system_access_users_utils.py

```diff
@@ -780,17 +832,25 @@ def add_or_update(self, user: User) -> None:
         )
         next_uid: Element = self.get("uid")
 
-        # since the current password of an user cannot not be compared with the new one,
-        # we're setting the password anyways
-        self.set_user_password(user)
-
         if existing_user:
+
+            if not password_verify(
+                existing_user_password=existing_user.password, password=user.password
+            ):
+                self.set_user_password(user)
+
+            # since we don't want the clear-type password to be set,
+            # and it is clear a update is not needed, we remove it from the update
+            if "password" in user.__dict__:
+                del user.__dict__["password"]
```

Comment:
Splitting the `password_hash` and `password` attributes would also affect this handling of the password attribute.